### PR TITLE
Convert process-config-entities to TypeScript

### PR DESCRIPTION
**Pull request by iantrich**

### CODE REVIEW

File: src/panels/lovelace/common/process-config-entities.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 // Parse array of entity objects from config
 import isValidEntityId from "../../../common/entity/valid_entity_id";
+import { EntityConfig } from "../entity-rows/types";
 
-export default function processConfigEntities(entities) {
+export const processConfigEntities = (entities: EntityConfig[]) => {
```
Comment (balloob) [inline src/panels/lovelace/common/process-config-entities.ts]: This is not correct. The output is `EntityConfig[]`, the input can also be a string (as tested for on line 19) so should be `Array<EntityConfig | string>`. Also please mark the return type

File: src/panels/lovelace/cards/hui-glance-card.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export class HuiGlanceCard extends hassLocalizeLitMixin(LitElement)
 
   public setConfig(config: Config): void {
     this._config = { theme: "default", ...config };
-    const entities = processConfigEntities(config.entities);
+    const entities = processConfigEntities(config.entities) as ConfigEntity[];
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-glance-card.ts]: We should have `processConfigEntities` take a type that you expect return values to be. 

(also, `ConfigEntity` and `EntityConfig`, we must really hate our future selves picking such confusing names)

Comment (balloob) [reply src/panels/lovelace/cards/hui-glance-card.ts]:
```suggestion
    const entities = processConfigEntities<ConfigEntity>(config.entities);
```

File: src/panels/lovelace/common/process-config-entities.ts (hunk 1 above)
Comment (balloob) [reply src/panels/lovelace/common/process-config-entities.ts]: I think that it should be something like this:

```suggestion
export const processConfigEntities = <T extends EntityConfig>(entities: Array<EntityConfig | string>[]): T[] => {
```

File: src/panels/lovelace/cards/hui-glance-card.ts (hunk 2 above)
Comment (iantrich) [reply src/panels/lovelace/cards/hui-glance-card.ts]: Yeah, reworking types for cards is on my todo list

**Comment by iantrich** [comment]
I'm not sure how to resolve the TypeScript issues that are now present after making the suggested changes:
image

**Comment by balloob** [comment]
Doesn't this work?

```ts
return entities.map((entityConf, index): T[] => {
```

**Comment by iantrich** [comment]
I believe this is related: https://github.com/palantir/tslint/issues/4239

**Review by balloob — APPROVED** [review]

**Pull request merged**